playbooks/byo/openshift-cluster/upgrades/v3_4/upgrade.yml

```diff
@@ -66,6 +66,10 @@
   tags:
   - pre_upgrade
 
+- include: ../../../../common/openshift-master/validate_restart.yml
```

Comment:
It's some logic that was pre-existing but was deep into the upgrade, it made sense on two fronts to push this further up into the upgrade before we make any real changes to validate the state of things, rather than letting it fail midway. 

The most valuable thing it does is make sure you're not trying to restart entire systems, while running ansible on one of them. (which would kill the whole upgrade)